decentraland/components/presentational/MyAssetsList.js

```diff
@@ -14,8 +14,15 @@ export default class MyAssetsList extends React.PureComponent {
     const { myAssets, userActions, renderItem } = this.props;
     const { length: listAmount } = myAssets;
 
+    const flatUserActions = Object.entries(userActions).map(userAction => {
+      const [actionId, userActionProps] = userAction;
```

Comment:
Simple question to make sure I understand: What's the structure of userAction in this line? And then remind me what this syntax on the left does to it?

Reply:
Structure: `userActions` = `{ 0x123789: { assetId, status }, 0x987321: { assetId, status} }` same as `creationActions` but having txhash as key instead of status.
It's nice to get action info from an id: `info = userActions[actionId]` (O(1) complexity).
But to get an action for a given asset (since that no loger has the `actionId`) a find is needed, to iterate over the `userActions` to get the one related to an asset, I've used `Object.entries(userActions)`  that returns an array of arrays where the first item is a property and the second item is the value of that property.
So, for the example object above we'll have:
```javascript
Object.entries({ 0x123789: { assetId, status }, 0x987321: { assetId, status} }
//=> [ ['0x123789', { assetId, status }], ['0x987321', { assetId, status}] ]
```
The syntax used here is to assign arrays values to some props.
```javascript
const [actionId, userActionProps] = userAction;
// is the same as
const actionId = userAction[0];
const userActionProps = userAction[1];
```

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

Reply:
> same as `creationActions` but having txhash as key instead of status

Oh right, status for `creationActions` is essentially a nickname for what the tx is. Which is nice unless the action needs to be retried, but even then there could be two action objects under that status nickname.

> The syntax used here is to assign arrays values to some props.

Got it - I was familar with that syntax, but the part I wasn't getting was what had turned it into an array like that. So your explanation of Object.entries is what I was missing. That was very helpful, thanks!

**Other thoughts**

Maybe a little of this belongs as a comment in the code? Comments in the code that say at a low level what the code does are pointless, but comments in the code about why a choice was made and the goal behind it are very helpful at preventing introduction of bugs or unnecessary rewrites the next time someone looks at this section.

So maybe a comment that says:

```js
// It's easy to look up action info from actionId
// But to get an action for a given asset (since the asset itself no longer has the actionId)
// a find is needed, to iterate over the userActions to get the one related to an asset,
// we used Object.entries to turn it into an array of arrays to help with this find
``

Reply:
Done.

Reply:
Thanks! Resolving this.